src/lib/containers/download_list/DownloadDetails.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from 'react'
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { faFile, faDatabase, faClock } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import calculateFriendlyFileSize from '../calculateFriendlyFileSize'
+import { testDownloadSpeed } from '../../utils/DownloadSpeedTest'
+import ReactTooltip from 'react-tooltip'
+import moment from 'moment'
+import { TOOLTIP_DELAY_SHOW } from '../table/SynapseTable'
+
+library.add(faFile)
+library.add(faDatabase)
+library.add(faClock)
+
+export type DownloadDetailsProps = {
+  numFiles: number
+  numBytes: number
+  token: string | undefined
+}
+
+type State = {
+  isLoading: boolean
+  downloadSpeed: number
+}
+
+export default function DownloadDetails(props: DownloadDetailsProps) {
+  const [state, setState] = useState<State>({
+    isLoading: true,
+    downloadSpeed: 0,
+  })
+  const { isLoading, downloadSpeed } = state
+  const { token, numFiles, numBytes } = props
+
+  useEffect(() => {
```

Comment:
If the intention to only run it on mount? 
If so,  may be this would be more explicit?
```useEffect(() => {
{
      if (token) {
        testDownloadSpeed(token).then(speed => {
          setState({
            isLoading: false,
            downloadSpeed: speed,
          })
        })
    }, []);
```
Or is this because there is a possibility that token will come through later?

Reply:
whoops! on mount was my intent, that would be better.